base/cvd/cuttlefish/host/commands/cvd/fetch/fetch_cvd.cc

```diff
@@ -475,6 +490,38 @@ Result<void> SaveConfig(FetcherConfig& config,
   return {};
 }
 
+Result<std::vector<std::string>> ExtractImageContents(
+    const std::string& image_filepath, const std::string& target_dir,
+    const bool keep_archive) {
+  if (!IsDirectory(image_filepath)) {
+    return CF_EXPECT(
+        ExtractArchiveContents(image_filepath, target_dir, keep_archive));
+  }
+
+  // The image is already uncompressed. Link or move its contents.
+  std::vector<std::string> files;
+  const std::function<bool(const std::string&)> file_collector =
+      [&files, &image_filepath,
+       &target_dir](const std::string& filepath) mutable {
+        std::string target_filepath =
+            target_dir + "/" + filepath.substr(image_filepath.size() + 1);
+        if (!IsDirectory(filepath)) {
+          files.push_back(target_filepath);
+        }
+        return true;
+      };
+  CF_EXPECT(WalkDirectory(image_filepath, file_collector));
+
+  if (keep_archive) {
+    // Must use hard linking due to the way fetch_cvd uses the cache.
+    CF_EXPECT(HardLinkDirecoryContentsRecursively(image_filepath, target_dir));
+  } else {
+    CF_EXPECT(MoveDirectoryContents(image_filepath, target_dir));
+    RecursivelyRemoveDirectory(image_filepath);
```

Comment:
`RecursivelyRemoveDirectory` can fail so it returns a `Result`. It's a valid choice to ignore that failure, but I wanted to confirm that this is intentional.

Reply:
Added a comment since you asked.